### PR TITLE
Help scroll top

### PR DESCRIPTION
**Pull request by ceford**
Pull Request for Issue #33722 .

### Summary of Changes

Added Javascript to scroll window to top on selection of Help page

### Testing Instructions

Go to Administrator / Help / Start Here - Joomla! Help
Scroll down to near the bottom of the long list and click an item there

Apply patch and rebuild with npm ci

### Actual result BEFORE applying this Pull Request

Something should happen, but
Nothing appears to happen because the Iframe with the new help page has been loaded out of sight.

### Expected result AFTER applying this Pull Request

The page will scroll to top bringing the newly loaded Help page into view.

### Documentation Changes Required

None

### CODE REVIEW

**Comment by Abernyte-Git** [comment]
I have tested this item :white_check_mark: successfully on 5348ac59ac0ff9a8e83cca271c3474bf68fd1046<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33742">issues.joomla.org/tracker/joomla-cms/33742</a>.</sub>

File: administrator/components/com_admin/tmpl/help/default.php [hunk 1]
```diff
@@ -16,6 +16,20 @@
 
 /** @var \Joomla\Component\Admin\Administrator\View\Help\HtmlView $this */
 
+/** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
+$wa = $this->document->getWebAssetManager();
+
+$wa->addInlineScript("
+document.addEventListener('DOMContentLoaded', function(event) {
```
Comment (dgrammatiko) [inline administrator/components/com_admin/tmpl/help/default.php]: Please use a js file

File: build/media_source/com_admin/js/admin-help.es6.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  'use strict';
+
+  document.addEventListener('DOMContentLoaded', function(event) {
+    let helpIndex = document.getElementById('help-index');
+    let links = helpIndex.querySelectorAll('a');
+    links && links.forEach(element => {
+      element.addEventListener('click', event => {
+        window.scroll(0,0);
+      });
+    })
```
Comment (dgrammatiko) [inline build/media_source/com_admin/js/admin-help.es6.js]: You can simplify it, to:

```js
/**
  * @copyright   (C) 2018 Open Source Matters, Inc. <https://www.joomla.org>
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
const helpIndex = document.getElementById('help-index');
if (helpIndex) {
  [].slice.call(helpIndex.querySelectorAll('a'))
    .map((element) => element.addEventListener('click', () => { window.scroll(0,0); }));
}
```

That's all the content of the file you need

**Comment by ceford** [comment]
So I moved the javascript to a file and am now looking at the build errors - not ready to retest.

**Comment by dgrammatiko** [comment]
> So I moved the javascript to a file and am now looking at the build errors - not ready to retest.

@ceford just use the snippet from https://github.com/joomla/joomla-cms/pull/33742#discussion_r629593166 should be CI friendly

**Comment by ceford** [comment]
@dgrammatiko  Thanks for the snippet - used that and now waiting for the checks to finish.

**Comment by dgrammatiko** [comment]
>  used that and now waiting for the checks to finish.

You need an empty line at the end

**Comment by dgrammatiko** [comment]
one more change `window.scroll(0,0);` to `window.scroll(0, 0);`

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on 05bad1f8309b1598e2312aacbf5dacf1a0ced8fe<br /><br />Note test instruction: rebuild js with `npm ci`<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33742">issues.joomla.org/tracker/joomla-cms/33742</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 05bad1f8309b1598e2312aacbf5dacf1a0ced8fe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33742">issues.joomla.org/tracker/joomla-cms/33742</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33742">issues.joomla.org/tracker/joomla-cms/33742</a>.</sub>

**Pull request merged**